src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,45 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        MachineProducer<?> someProducer;
+        if (type == Bulldozer.class) {
+            someProducer = new BulldozerProducer();
+        } else if (type == Excavator.class) {
+            someProducer = new ExcavatorProducer();
+        } else if (type == Truck.class) {
+            someProducer = new TruckProducer();
+        } else {
+            return new ArrayList<>();
+        }
```

Comment:
```suggestion
        }
```

Reply:
I cannot understand this change. The last Else block is for returning the empty list in the case when no producer found. 
If this block is removed, the compile-time error is appeared that "Variable 'producer' might not have been initialized". If I initialize it with null, then NPE throws in the case no producer found.

Reply:
I didn't like that in the last if you return something and in the others not, it'd be cool if you rewrite it. For example, assign value for machines directly in ifs not to producer then you can remove 26th line